### PR TITLE
Create Subarray sum

### PR DESCRIPTION
**Pull request by shubham1710**

### CODE REVIEW

**Review by abhishekchopra13 — CHANGES_REQUESTED** [review]
Check the sample code for implementation. There is no need for description of the process. Code should be easy to implement and must act as a flexible template.

File: Subarray sum [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+
+
+Explanation 2:
+
+No subarray of elements sum up to give 8.
+
+As 4+5=9, 4+5+1=10, 5+1=6, none matches 8.
+
+
```
Comment (Bhavnaharitsa) [inline Subarray sum:79]: One explanation is enough if you want to give a explanation

File: Subarray sum [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    /* Add elements one by one to curr_sum and 
+
+    if the curr_sum exceeds the sum,
+
+    then remove starting element */
+
+    for (i = 1; i <= n; i++)
+
+    {
```
Comment (Bhavnaharitsa) [inline Subarray sum:116]: Too many comments

File: Subarray sum [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    subArraySum(arr, n, sum);
+
+    return 0;
+
+}
+
+
+
+Solution Explanation:
+
```
Comment (Bhavnaharitsa) [inline Subarray sum:199]: Too much explanation, the code should be explanatory. Please remove all the  explanation that are not needed

**Comment by chanmol1999** [comment]
No Response going to close

**Pull request closed without merging**